lib/rspec/expectations/configuration.rb

```diff
@@ -141,14 +149,43 @@ def self.format_backtrace(backtrace)
       # Configures whether RSpec will warn about matcher use which will
       # potentially cause false positives in tests.
       #
-      # @param value [Boolean]
-      attr_writer :warn_about_potential_false_positives
+      # @param [Boolean] boolean
+      def warn_about_potential_false_positives=(boolean)
+        if boolean
+          self.on_potential_false_positives = :warn
+        elsif warn_about_potential_false_positives?
```

Comment:
`warn_about_potential_false_positives = false` controls wether we issue a warning or not on encountering a potential false positive. If we are raising an exception when encountering a potential false positive we are not issuing a warning, so we don't need to change anything.